src/utils/config.go

```diff
@@ -0,0 +1,60 @@
+package utils
+
+import (
+	"encoding/json"
+	"errors"
+	"log/slog"
+	"slices"
+	"time"
+)
+
+var allowedEncodingList = []string{"ISO-8859-1", "UTF-8"}
+
+//https://www.notion.so/flexion-cdc-ti/Thinking-about-config-eb9424dafea14320be5cee1b8b03d890?pvs=4
+
+// All the details each partner should have
+// to null-equivalent, but we'll want to check for valid values
+type PartnerSettings struct {
+	DisplayName              string `json:"displayName"` // full name if we need pretty names
+	IsActive                 bool   `json:"isActive"`
+	IsExternalSftpConnection bool   `json:"isExternalSftpConnection"`
+	HasZipPassword           bool   `json:"hasZipPassword"`
+	DefaultEncoding          string `json:"defaultEncoding"`
+}
+
+// When did we last get the file and what did we parse out of it?
+type Config struct {
+	lastRetrieved   time.Time
+	partnerSettings PartnerSettings
```

Comment:
Just checking that you are purposely making these private fields (due to the lowercase start of the name).

Reply:
I think for now these settings are okay